Item 8357: LKFM Inventory data folder archive export/import

cnathe opened the pull request:
#### Rationale
As a precursor to allowing for LKSM/FM evaluation data to be imported via a folder archive, this story supports the inventory data part of the LKFM app. This PR adds the folder archive data type for "Inventory data" to the folder.xsd and list of FolderArchiveDataTypes.

#### Related Pull Requests
* https://github.com/LabKey/dataintegration/pull/86
* https://github.com/LabKey/platform/pull/1917
* https://github.com/LabKey/inventory/pull/174

#### Changes
* Add "inventory" export dir type to folder.xsd for folder.xml
* Add INVENTORY type to FolderArchiveDataTypes
* Minor updates to SampleTypeAndDataClassFolderImporter for consistency with other importers

cnathe committed:
Item 8357: Inventory module data writer
- initial implementation that writes out tsv for inventory schema visible tables
- and placeholder for importer class

cnathe committed:
Merge branch 'develop' into fb_inventoryWriterImporter

cnathe committed:
Merge branch 'develop' into fb_inventoryWriterImporter

cnathe committed:
SampleTypeAndDataClassFolderImporter TODO for explaining why priority is set

cnathe committed:
Add inventory exportDirType to folder.xml

cnathe committed:
Update job status for inventory and sampletype/dataclass importers

cnathe committed:
Remove TODO about INVENTORY data type name

cnathe committed:
Merge branch 'develop' into fb_inventoryWriterImporter

cnathe commented on experiment/src/org/labkey/experiment/samples/SampleTypeAndDataClassFolderImporter.java:
```diff
@@ -239,6 +242,7 @@ public FolderImporter create()
         @Override
         public int getPriority()
         {
+            // TODO add a note about why this value is set as such? what other importer does it need to be after?
```
@labkey-klum I was looking at this folder importer while doing the dev for the LKFM inventory data importer. I noticed that this importer priority was set, but I wasn't sure why. Can you help me fill in this TODO about which importer we are making sure happens before the sample type and data class import?

labkey-klum replied:
I set this so that it would run after the experiments & runs importer, both importers use the XAR format to capture the lineage for sample types and data classes and I can't remember the exact problem but I think it was related to having an assay run with references to sample type information, wouldn't get hooked up correctly if the order was inverted.

cnathe replied:
That makes sense. I'll update that comment accordingly. Thanks.

cnathe committed:
Update SampleTypeAndDataClassFolderImporter importer priority comment

cnathe committed:
Merge branch 'develop' into fb_inventoryWriterImporter

cnathe committed:
Merge branch 'develop' into fb_inventoryWriterImporter

cnathe committed:
Misc fixes/changes after local testing

cnathe committed:
Fix for optimistic concurrency exception on folder import when domain is deleted and then recreated
- domain related caches were being cleared but no the domain property caches
- this happens for the inventory.Location table when the inventory data is being replaced

cnathe committed:
Fix up importer error messages

cnathe committed:
Merge branch 'develop' into fb_inventoryWriterImporter

cnathe committed:
Merge branch 'develop' into fb_inventoryWriterImporter

cnathe committed:
Fix for optimistic concurrency exception for inventory data import
